### PR TITLE
Schedule, Schedule entries, Trigger, Acknowledge, and Log Entries

### PR DESCRIPTION
**Pull request by lszomoru**
I was wondering if you would be willing to accept my pull request which contains wrappers for the Schedule, Schedule entries, Trigger, and Acknowledge REST APIs in order to extend the capabilities of the library. If you are willing to accept the pull request it would be great if you could post an updated package to Nuget. Do let me know if there are any changes to the code that you want me to make in order to accept the pull request. Thanks!

### CODE REVIEW

**Comment by usbsnowcrash** [comment]
I'm traveling this weekend but should be able to look Sunday or Monday

**Comment by lszomoru** [comment]
No worries. Actually I just realized that I would like to add another REST API wrapper for getting the log entries for a particular incident. Let me work on that and I will update the pull request.

**Comment by lszomoru** [comment]
I have added the wrappers for the Log Entries. Please review the code and let me know your feedback.  Thanks!

File: Source/PagerDuty.Net.Tests/Models/IncidentsResponseTests.cs [hunk 1]
```diff
@@ -30,12 +30,12 @@ public class IncidentsResponseTests {
 
             Assert.IsNotNull(myAlert);
             Assert.IsNotNull(myAlert.incidents);
+            Assert.IsNotNull(myAlert.incidents);
```
Comment (usbsnowcrash) [inline Source/PagerDuty.Net.Tests/Models/IncidentsResponseTests.cs]: Why do we assert this twice?

File: Source/PagerDuty.Net.Tests/Models/AlertTests.cs [hunk 2]
```diff
@@ -33,7 +33,7 @@ public class AlertTests {
             Assert.AreEqual("PWL7QXS", myAlert.id);
             Assert.AreEqual("Phone", myAlert.type);
             Assert.AreEqual("+15555551234", myAlert.address);
-            Assert.AreEqual(new DateTime(634981805300000000), myAlert.started_at);
+            Assert.AreEqual(new DateTime(634982021300000000), myAlert.started_at);
```
Comment (usbsnowcrash) [inline Source/PagerDuty.Net.Tests/Models/AlertTests.cs]: I didn't realize my test was failing in master!  thanks!

File: Source/PagerDuty.Net/Models/Channel.cs [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+namespace PagerDuty.Net {
+    [Serializable()]
+    public class Channel {
+        //Default fields
+        public string type { get; set; }
+        public string summary { get; set; }
+
+        //Channel specific fields
+
+        //Nagios, API, Web trigger channel specific
```
Comment (usbsnowcrash) [inline Source/PagerDuty.Net/Models/Channel.cs]: What does Nagios mean?

File: Source/PagerDuty.Net/Models/Channel.cs [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        public object details { get; set; }
+
+        //Nagios 
+        public string host { get; set; }
+        public string service { get; set; }
+        public string state { get; }
```
Comment (usbsnowcrash) [inline Source/PagerDuty.Net/Models/Channel.cs]: needs bot a get and a set

File: Source/PagerDuty.Net/PagerDutyAPI.cs [hunk 5]
```diff
@@ -228,6 +346,80 @@ public class PagerDutyAPI {
...
+            return resp.Data;
+        }
+
+        /// <summary>
+        /// Trigger an incident using the integration API.
+        /// </summary>
+        /// <param name="integrationUrl">Integration API Url</param>
+        /// <param name="aRequest">Trigger details</param>
+        /// <returns></returns>
+        public IntegrationResponse Trigger(string integrationUrl, TriggerRequest tRequest) {
```
Comment (usbsnowcrash) [inline Source/PagerDuty.Net/PagerDutyAPI.cs]: Why have them pass in the integrationurl?  When you instantiate the PagerDuty api wrapper you pass in the subdomain pointing to your account.  If you look at this https://github.com/usbsnowcrash/PagerDuty.Net/blob/master/Source/PagerDuty.Net/PagerDutyAPI.cs#L34 you can see how I do it for the other endpoints.

File: Source/PagerDuty.Net/PagerDutyAPI.cs [hunk 6]
```diff
@@ -47,6 +47,16 @@ public class PagerDutyAPI {
         }
 
         /// <summary>
+        /// Acknowledge an incident using the integration API.
+        /// </summary>
+        /// <param name="integrationUrl">Integration API Url</param>
+        /// <param name="aRequest">Acknowledge details</param>
+        /// <returns></returns>
+        public IntegrationResponse Acknowledge(string integrationUrl, AcknowledgeRequest aRequest) {
```
Comment (usbsnowcrash) [inline Source/PagerDuty.Net/PagerDutyAPI.cs]: Passing in an integration url does not make sense I think.  It looks like (from the pagerduty api docs) that you always post to https://events.pagerduty.com/generic/2010-04-15/create_event.json for the integration apis.

File: Source/PagerDuty.Net.Tests/Models/IncidentsResponseTests.cs (hunk 1 above)
Comment (lszomoru) [inline Source/PagerDuty.Net.Tests/Models/IncidentsResponseTests.cs]: Fixed.

File: Source/PagerDuty.Net.Tests/Models/AlertTests.cs (hunk 2 above)
Comment (lszomoru) [inline Source/PagerDuty.Net.Tests/Models/AlertTests.cs]: No problem!

File: Source/PagerDuty.Net/Models/Channel.cs (hunk 3 above)
Comment (lszomoru) [inline Source/PagerDuty.Net/Models/Channel.cs]: https://www.pagerduty.com/docs/guides/nagios-integration-guide/

**Comment by lszomoru** [comment]
Not really sure about the AppVeyor build failure, as the two tests that fail are passing on the latest code.

File: Source/PagerDuty.Net/Models/Channel.cs (hunk 4 above)
Comment (lszomoru) [inline Source/PagerDuty.Net/Models/Channel.cs]: Fixed.

File: Source/PagerDuty.Net/PagerDutyAPI.cs (hunk 6 above)
Comment (lszomoru) [inline Source/PagerDuty.Net/PagerDutyAPI.cs]: Fixed.

**Comment by usbsnowcrash** [comment]
I see the same tests failing on my machine as they do in appveyor.

screenshot 2015-07-20 14 08 23
I think we need to change the asserts in these two tests to be something like this
```cs
Assert.AreEqual(DateTime.Parse("2013-03-06T15:28:50-05:00"), myAlert.started_at);
```
Once these are passing I will merge :zap:

**Comment by lszomoru** [comment]
Strange. Let me take a look and update the pull request.

**Comment by lszomoru** [comment]
Fixed. I think I know what the problem was. Those two tests were using JSON files that had time zone specific timestamps in them. Due to that the tests were failing on my machine. After I have fixed them, they were passing on my machine but were failing on the build and your machine. The fix was to switch to a UTC timestamp.

**Comment by lszomoru** [comment]
Thanks for merging in my changes. Are you planning on publishing the new version to Nuget?

**Comment by usbsnowcrash** [comment]
@lszomoru already published, just takes a few min for it to show up https://www.nuget.org/packages/PagerDutyNet/

**Comment by lszomoru** [comment]
@usbsnowcrash, awesome! Thanks! Please feel free to ping me if anyone reports any issues with the newly added wrappers. I will continue to look at PagerDuty and will post a new pull request in case I am adding new functionality.

**Pull request merged**